infra/base-images/base-runner/test_all

```diff
@@ -44,14 +54,73 @@ for FUZZER_BINARY in $(find $OUT/ -executable -type f); do
     fi
   fi
 
-  bad_build_check $FUZZER_BINARY
-  N=$[$N+1]
+  # Launch bad build check process in the background.
+  bad_build_check $FUZZER_BINARY $VALID_TARGETS_DIR $BROKEN_TARGETS_DIR &
+
+  # Count total number of fuzz targets being tested.
+  TOTAL_TARGETS_COUNT=$[$TOTAL_TARGETS_COUNT+1]
+
+  # Do not spawn more processes than the number of CPUs available.
+  n_child_proc=$(jobs -p | wc -l)
+  while [ "$n_child_proc" -eq "$NPROC" ]; do
+    sleep 4
+    n_child_proc=$(jobs -p | wc -l)
+  done
 done
 
-if [ "$N" -eq "0" ]; then
+# Wait for background processes to finish.
+wait
+
+# Sanity check in case there are no fuzz targets in the $OUT/ dir.
+if [ "$TOTAL_TARGETS_COUNT" -eq "0" ]; then
   echo "ERROR: no fuzzers found in $OUT/"
   ls -al $OUT
   exit 1
 fi
 
-echo "$N fuzzers total"
+# Calculate number of valid and broken fuzz targets.
+VALID_TARGETS_COUNT=$(ls $VALID_TARGETS_DIR | wc -l)
+BROKEN_TARGETS_COUNT=$(ls $BROKEN_TARGETS_DIR | wc -l)
+
+# Sanity check to make sure that bad build check doesn't skip any fuzz target.
+if [ "$TOTAL_TARGETS_COUNT" -ne "$[$VALID_TARGETS_COUNT+$BROKEN_TARGETS_COUNT]" ]; then
+  echo "ERROR: bad_build_check seems to have a bug, total number of fuzz" \
+       "does not match number of fuzz targets tested."
+  echo "Total fuzz targets ($TOTAL_TARGETS_COUNT):"
+  ls -al $OUT
+  echo "Valid fuzz targets ($VALID_TARGETS_COUNT):"
+  ls -al $VALID_TARGETS_DIR
+  echo "Total fuzz targets ($BROKEN_TARGETS_COUNT):"
+  ls -al $BROKEN_TARGETS_DIR
+  exit 1
+fi
+
+# Build info about all broken fuzz targets (if any).
+if [ "$BROKEN_TARGETS_COUNT" -gt "0" ]; then
+  echo "Broken fuzz targets ($BROKEN_TARGETS_COUNT):"
+  for target in $(ls $BROKEN_TARGETS_DIR); do
+    echo "${target}:"
+    cat ${BROKEN_TARGETS_DIR}/${target}
+  done
+fi
+
+# Calculate the percentage of broken fuzz targets and make the finel decision.
+BROKEN_TARGETS_PERCENTAGE=$[$BROKEN_TARGETS_COUNT*100/$TOTAL_TARGETS_COUNT]
+
+
+if [ "$BROKEN_TARGETS_PERCENTAGE" -gt "90" ]; then
+  echo "ERROR: $BROKEN_TARGETS_PERCENTAGE% of fuzz targets seem to be broken." \
+       "See the list above for a detailed information."
+
+  # TODO: figure out how to not fail the "special" cases handled below. Those
+  # are from "example" and "c-ares" projects and are too small targets to pass.
+  if [ "(ls $OUT/do_stuff_fuzzer $OUT/ares_*_fuzzer | wc -l)" -gt "0" ]; then
```

Comment:
for these fuzz targets the output would be looking like:

```
INFO: performing bad build checks for /out/ares_create_query_fuzzer
INFO: performing bad build checks for /out/ares_parse_reply_fuzzer
INFO: /out/ares_parse_reply_fuzzer has passed bad build checks.
INFO: /out/ares_create_query_fuzzer has failed bad build checks.
Broken fuzz targets (1):
ares_create_query_fuzzer:
BAD BUILD: the target seems to have only partial coverage instrumentation.
2 fuzzers total, 1 seem to be broken (50%).
root@f8556f4dea8b:/out# echo $?
0
```

i.e. not breaking the build

Reply:
That's a great suggestion, but I don't seem to be able to find a good formula for that, see:

```
arduinojson
 12728 KB, 407 PCs

c-ares
 12656 KB, 42 PCs
 12772 KB, 459 PCs

bad_example valid build
 12764 KB, 662 PCs

example
 12664 KB, 73 PCs

libpng
 13304 KB, 4471 PCs

ffmpeg, any target:
118216 KBs, 292815 PCs
```

I'd say that we have to treat cases under the threshold as buggy targets and handle them on a case by case basis. Another option would be to remove `ares_create_query_fuzzer` as 42 PCs is a very small number

Reply:
Wasn't `ares_create_query_fuzzer` added in response to its use in a ChromeBook full-chain exploit? Losing that coverage seems unfortunate.

Reply:
That's true, and this is the only reason that prevents me from deleting it :))